packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -189,7 +189,9 @@ export const LightningElement: LightningElementConstructor = function (
 ): LightningElement {
     // This should be as performant as possible, while any initialization should be done lazily
     if (isNull(vmBeingConstructed)) {
-        throw new ReferenceError('Illegal constructor');
+        // TypeError matches the spec for an illegal custom element constructor:
```

Comment:
I'm not sure this message is accurate, it is a illegal constructor because LWC can't make sense of it, has nothing to do to CE or alike.

Reply:
You're right; I don't think it has anything to do with CEs. But I do think TypeError is probably more appropriate here than ReferenceError.

I'll change the comment